code/game/objects/structures/noticeboard.dm

```diff
@@ -1,40 +1,52 @@
+#define MAX_NOTICES 8
+
 /obj/structure/noticeboard
 	name = "notice board"
 	desc = "A board for pinning important notices upon."
 	icon = 'icons/obj/stationobjs.dmi'
-	icon_state = "nboard00"
-	density = 0
-	anchored = 1
+	icon_state = "noticeboard"
+	density = FALSE
+	anchored = TRUE
+	integrity_max = 150
+	/// Current number of a pinned notices
 	var/notices = 0
 
 /obj/structure/noticeboard/Initialize(mapload, dir, building = FALSE)
+	. = ..()
+
 	if(building)
 		pixel_x = (dir & 3)? 0 : (dir == 4 ? -32 : 32)
 		pixel_y = (dir & 3)? (dir ==1 ? -27 : 27) : 0
-		update_icon()
-	if(mapload)
-		for(var/obj/item/I in loc)
-			if(notices > 4)
-				break
-			if(istype(I, /obj/item/paper))
-				I.forceMove(src)
-				notices++
-	icon_state = "nboard0[notices]"
-	. = ..()
+		update_appearance(UPDATE_ICON)
+
+	if(!mapload)
+		return
+
+	for(var/obj/item/I in loc)
+		if(notices > MAX_NOTICES)
+			break
+		if(istype(I, /obj/item/paper))
+			I.forceMove(src)
+			notices++
+	update_appearance(UPDATE_ICON)
 
 //attaching papers!!
-/obj/structure/noticeboard/attackby(var/obj/item/O as obj, var/mob/user as mob)
-	if(istype(O, /obj/item/paper))
-		if(notices < 5)
-			if(!user.attempt_insert_item_for_installation(O, src))
+/obj/structure/noticeboard/attackby(obj/item/O, mob/user, params)
```

Comment:
just making a note that this doesn't adhere to the current proc signature of attackby(), but since it's updating legacy code to effectively to-be-changed-later code (we'll want to do persistence and other stuff later) i'm gonna just ignore it